ToolBox/SECDataDownloader/SECDataConverter.cs

```diff
@@ -123,189 +116,189 @@ public SECDataConverter(string rawSource, string destination, string knownEquity
                         }
                     }
                 );
-
-            KnownEquities = Directory.GetFiles(knownEquityFolder)
-                .Select(x => Path.GetFileNameWithoutExtension(x).ToLower())
-                .ToList();
         }
 
         /// <summary>
         /// Converts the data from raw format (*.nz.tar.gz) to json files consumable by LEAN
         /// </summary>
         /// <param name="startDate">Starting date to start process files</param>
         /// <param name="endDate">Ending date to stop processing files</param>
-        public void Process(DateTime startDate, DateTime endDate)
+        public void Process(DateTime processingDate)
         {
+            var formattedDate = processingDate.ToString(DateFormat.EightCharacter);
+            var remoteRawData = new FileInfo(Path.Combine(RawSource, $"{formattedDate}.nc.tar.gz"));
+            if (!remoteRawData.Exists)
+            {
+                throw new Exception($"SECDataConverter.Process(): Raw data {remoteRawData} not found. No process can be done.");
+            }
+
+            Log.Trace($"SECDataConverter.Process(): Copying raw data locally...");
+
+            var localRawData = remoteRawData.CopyTo(Path.Combine(Path.GetTempPath(), remoteRawData.Name));
+            var extractDataPath = Path.Combine(Path.GetTempPath(), formattedDate);
+
+            Log.Trace($"SECDataConverter.Process(): Extract raw data...");
+
+            using (var data = localRawData.OpenRead())
+            {
+                using (var archive = TarArchive.CreateInputTarArchive(new GZipInputStream(data)))
+                {
+                    Directory.CreateDirectory(extractDataPath);
+                    archive.ExtractContents(extractDataPath);
+
+                    Log.Trace($"SECDataConverter.Process(): Extracted SEC data to path {extractDataPath}");
+                }
+            }
+
+            var tickerList = _tickerFolder.EnumerateDirectories().AsParallel()
+                .Where(d => d.EnumerateFiles($"{formattedDate}*").Any())
+                .Select(d => d.Name);
```

Comment:
OK, so this might be unexpected, but this `tickerList` _enumerable_ is used in exactly one location, [line 266](https://github.com/QuantConnect/Lean/pull/3348/files#diff-415f3d17a5d03708870daca21d68f5e5R266) which is invoked inside a loop _and_ inside a `Parallel.ForEach` block (that's n^2 invocations. Add that to the fact that this is an _enumerable_ and each time you call `tickerList.Contains` it will re-enumerate the enumerable until it finds a matching element, and if no element is found, then it will enumerate the entire enumerable, _including_  enumeration of `EnumerateFiles` defines on [line 154](https://github.com/QuantConnect/Lean/pull/3348/files#diff-415f3d17a5d03708870daca21d68f5e5R154). The easy way to remedy this is to call `ToHashSet()` at the end which will cause the enumerable to be materialized. _Further_, any time you're using a collection _only_ to call it's `Contains` method, you should be using a `HashSet<T>` as it provides `O(1)` (constant time) lookups, vs `O(n)` for `IEnumerable<T>`.

So this loop w/ the contains call is currently executing on `O(n^3)`  !!!!!